### PR TITLE
Move documentation from README.md into book structure

### PR DESCRIPTION
**Pull request by bbros-dev**
Close issue #317

### CODE REVIEW

File: book/book.toml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+title = "The Goose Book"
+authors = ["Jeremy Andrews"]
+description = "Load testing framework, inspired by Locust, implemented with Rust."
+language = "en"
+
+[rust]
+edition = "2018"
+
+[build]
+build-dir = "html"
```
Comment (jeremyandrews) [inline book/book.toml:11]: Do not override, leave the default.

File: book/book.toml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+create-missing = true
```
Comment (jeremyandrews) [inline book/book.toml:12]: Do not override, leave the default: I'd prefer the build generates an error instead of creating empty pages.

File: book/src/404.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# Document not found (404)
+
+Apologies: This URL is invalid. Try the search instead.
```
Comment (jeremyandrews) [inline book/src/404.md:3]: This page does not appear to be used, remove.

File: book/src/SUMMARY.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+- [Defaults](defaults.md)
+- [Controlling Running Goose Load Test](controlling-running-goose-load-test.md)
+- [Throttling Requests](throttling-requests.md)
+- [Logging Load Test Errors](logging-load-test-errors.md)
+- [Logging Load Test Requests](logging-load-test-requests.md)
+- [Logging Load Test Tasks](logging-load-test-tasks.md)
+- [Load Test Debug Logging](load-test-debug-logging.md)
+- [Coordinated Omission Mitigation](coordinated-omission-mitigation.md)
+
+# Gaggle
```
Comment (jeremyandrews) [inline book/src/SUMMARY.md:19]: Do not add/change the structure with this first PR. Remove the #Gaggle header here.

File: book/src/contributors.md [hunk 5]
```diff
@@ -0,0 +1,15 @@
+# Contributors
```
Comment (jeremyandrews) [inline book/src/contributors.md:1]: Remove this file, it is duplicate and will be difficult to keep up to date.

File: book/src/developers.md [hunk 6]
```diff
@@ -0,0 +1,3 @@
+# Developers
+
+[Developer documentation (external link)](https://docs.rs/goose/)
```
Comment (jeremyandrews) [inline book/src/developers.md:3]: Remove, appropriate links will be added in followup when we focus on properly structuring this book.

File: book/src/gaggle.md [hunk 7]
```diff
@@ -0,0 +1 @@
+# Gaggle
```
Comment (jeremyandrews) [inline book/src/gaggle.md:1]: Remove

File: book/src/goose.md [hunk 8]
```diff
@@ -0,0 +1 @@
+# Goose
```
Comment (jeremyandrews) [inline book/src/goose.md:1]: This page seems unused?

File: examples/the-internet/ .dockerignore [hunk 9]
```diff
@@ -0,0 +1,2 @@
+/target/
+/.git/
```
Comment (jeremyandrews) [inline examples/the-internet/ .dockerignore:2]: a series of unrelated files, remove

File: book/src/tips.md [hunk 10]
```diff
@@ -0,0 +1,7 @@
+# Tips
+
+* Avoid `unwrap()` in your task functions -- Goose generates a lot of load, and this tends
+to trigger errors. Embrace Rust's warnings and properly handle all possible errors, this
+will save you time debugging later.
+* When running your load test for real, use the cargo `--release` flag to generate
+optimized code. This can generate considerably more load test traffic.
```
Comment (jeremyandrews) [inline book/src/tips.md:7]: This page is orphaned, it needs a link from the Overview.

File: .github/workflows/CI.yml [hunk 11]
```diff
@@ -7,29 +7,52 @@ on:
...
+        run: mdbook build book && mdbook test book
+      - name: Deploy Goose book
+        uses: peaceiris/actions-gh-pages@v3.5.4-6
+        # Workaround https://github.com/peaceiris/actions-gh-pages/issues/627
+        env:
+          ACTIONS_ALLOW_UNSECURE_COMMANDS: true
+        with:
+          deploy_key: ${{ secrets.ACTIONS_DEPLOY_KEY }}
+          publish_dir: ./book/html
+          publish_branch: gh-pages
```
Comment (jeremyandrews) [inline .github/workflows/CI.yml:58]: I would prefer to deal with automatically building and deploying the book in a followup. (So do not make changes to this file in this PR.)

File: book/book.toml [hunk 12]
```diff
@@ -0,0 +1,15 @@
...
+
+[output.html]
+mathjax-support = true
```
Comment (jeremyandrews) [inline book/book.toml:15]: This is an unused feature at this time, remove.

File: .gitignore [hunk 13]
```diff
@@ -4,3 +4,4 @@
 /settings.json.code-workspace
 **swp
 /Cargo.lock
+book/html
```
Comment (jeremyandrews) [inline .gitignore:7]: this will need to be updated to reflect the proper book location (`src/docs/goose-book/book`)

File: README.md [hunk 14]
```diff
@@ -21,893 +17,3 @@ Goose is a Rust load testing tool inspired by [Locust](https://locust.io/). User
   - [Real-life load testing with Goose](https://www.tag1consulting.com/blog/real-life-goose-load-testing)
   - [Gaggle: a distributed load test](https://www.tag1consulting.com/blog/show-me-how-flock-flies-working-gaggle-goose)
   - [Optimizing Goose performance](https://www.tag1consulting.com/blog/golden-goose-egg-compile-time-adventure)
```
Comment (jeremyandrews) [inline README.md:19]: Add some simple documentation here about how to build the book, for example:

> Manually build and view "The Goose Book" with [mdBook](https://rust-lang.github.io/mdBook/) with the following command:
> ```
>     mdbook build src/docs/goose-book --open
> ```

**Review by jeremyandrews — CHANGES_REQUESTED** [review]
This is very close, thanks! As noted in the above comments:

- create the book in `src/docs/goose-book/` instead of `book/`
- do not auto-build and auto-deploy the book with this PR
- do not create empty filler pages
- remove the unrelated example from this PR
- do not add/change the documentation structure with this PR
- HOWEVER: do add simple documentation on how to manually build the documentation

File: book/src/overview.md [hunk 15]
```diff
@@ -0,0 +1,7 @@
+# Overview
+
+Goose is a Rust load testing tool inspired by [Locust](https://locust.io/). User behavior is defined with standard Rust code. Load tests are applications that have a dependency on the Goose library. Web requests are made with the [Reqwest](https://docs.rs/reqwest) HTTP Client.
+
+## Requirements
+
+Minimum required `rustc` version is `1.49.0`: `goose` depends on [`flume`](https://docs.rs/flume) for communication between threads, which in turn depends on [`spinning_top`](https://docs.rs/spinning_top) which uses `hint::spin_loop` which stabilized in `rustc` version `1.49.0`. More detail in https://github.com/rust-lang/rust/issues/55002.
```
Comment (jeremyandrews) [inline book/src/overview.md:7]: Leave `Requirements` in README.md.

Comment (bbros-dev) [reply book/src/overview.md:7]: We believe you can merge then revert 12d0338.  Resolving as we can't revise within 3-6 months (see comment below).

**Comment by bbros-dev** [comment]
FYI, our time budget is exhausted, and we can't see returning to this for the foreseeable future (3-6 months).  Happy for anyone else to do as they wish with this PR.

File: README.md (hunk 14 above)
Comment (bbros-dev) [reply README.md:19]: We opened a RFC and there was discussion around a skeleton when this wasn't mentioned as 'nice to have' additions. Resolving as can't fix in 3-6 months.

File: .gitignore (hunk 13 above)
Comment (bbros-dev) [reply .gitignore:7]: Resolving as can't fix in 3-6 months.

File: book/src/tips.md (hunk 10 above)
Comment (bbros-dev) [reply book/src/tips.md:7]: Resolving as can't fix in 3-6 months.

File: book/src/SUMMARY.md (hunk 4 above)
Comment (bbros-dev) [reply book/src/SUMMARY.md:19]: Resolving as can't fix in 3-6 months.

File: book/src/404.md (hunk 3 above)
Comment (bbros-dev) [reply book/src/404.md:3]: Resolving as can't fix in 3-6 months.

File: book/book.toml (hunk 2 above)
Comment (bbros-dev) [reply book/book.toml:12]: Resolving as can't fix in 3-6 months.

File: book/book.toml (hunk 1 above)
Comment (bbros-dev) [reply book/book.toml:11]: Resolving as can't fix in 3-6 months.

File: README.md [hunk 16]
```diff
@@ -8,10 +8,6 @@ Have you ever been attacked by a goose?
 [![CI](https://github.com/tag1consulting/goose/workflows/CI/badge.svg)](https://github.com/tag1consulting/goose/actions?query=workflow%3ACI)
 [![Docker Repository on Quay](https://quay.io/repository/tag1consulting/goose/status "Docker Repository on Quay")](https://quay.io/repository/tag1consulting/goose)
 
-## Overview
-
-Goose is a Rust load testing tool inspired by [Locust](https://locust.io/). User behavior is defined with standard Rust code. Load tests are applications that have a dependency on the Goose library. Web requests are made with the [Reqwest](https://docs.rs/reqwest) HTTP Client.
-
```
[the remark this answers is not in the document]
Comment (bbros-dev) [reply README.md:14]: Resolving as can't fix in 3-6 months.

**Comment by jeremyandrews** [comment]
Replaced by #361
